source/PlayerInfo.cpp

```diff
@@ -1865,6 +1865,21 @@ Mission *PlayerInfo::BoardingMission(const shared_ptr<Ship> &ship)
 
 
 
+bool PlayerInfo::CaptureOverriden(const shared_ptr<Ship> &ship) const
+{
+	if(ship->IsCapturable())
+		return false;
+	const Mission *mission = boardingMissions.empty() ? nullptr : &boardingMissions.back();
+	// Allow trying to board the ship again once it's been allowed once.
+	if(!mission)
+		for(const Mission &mission : Missions())
+			if(mission.OverridesCapture() && !mission.IsFailed() && mission.SourceShip() == ship.get())
+				return true;
+	return mission && mission->OverridesCapture() && !mission->IsFailed() && mission->SourceShip() == ship.get();
```

Comment:
I feel like we could use a little extra commenting here.
```suggestion
	// Check if there's a boarding mission being offered which allows this ship to be captured. If the boarding
	// mission was declined, then this results in one-time capture access to the ship, If it was accepted, then
	// the next boarding attempt will have the boarding mission in the player's active missions list, checked below.
	const Mission *mission = boardingMissions.empty() ? nullptr : &boardingMissions.back();
	// Otherwise, check if there's an already active boarding mission. This allows trying to board the ship again after accepting the mission.
	if(!mission)
		for(const Mission &mission : Missions())
			if(mission.OverridesCapture() && !mission.IsFailed() && mission.SourceShip() == ship.get())
				return true;
	return mission && mission->OverridesCapture() && !mission->IsFailed() && mission->SourceShip() == ship.get();
```